File: ch9/src/Bird.py
from Noise import Noise


class Bird(Noise):
    def __init__(self):
        pass
    
    def move_noise(self):
        return ("*flap flap flap flap flap flap*")
    def honk(self):
        return("*tweedee-dee tweeedeee-deee*")
        
    def lay_egg(self):
        return(str(self) + " laid an egg")

    def __str__(self) -> str:
        return ("lil burb")
        

File: ch8/src/Bird.java
/******************************************************************************
 Patrick Gilfether
 Midterm Synthesis, ch.8 practical
 CS 5004
 10/13/2022
 *******************************************************************************/

//This class exists solely to demonstrate a non-vehicle implementation of Noise interface
public class Bird implements Noise{
    @Override
    public void driveNoise(){
        System.out.println("*flap flap flap flap flap flap*");
    }

    @Override
    public void honk() {
        System.out.println("*tweedeee-dee... tweeedeee-dee...*");
    }
}
